### PR TITLE
Skynet: award team medals

### PR DESCRIPTION
**Pull request by SimpleTease**

### CODE REVIEW

**Comment by SimpleTease** [comment]
It's most noticed on Skynet that all team players get medals.

Also players can keep switching rules after every round. This creates visual bugs on award screen.

White 1 = 2 medals
White 2 = 4 medals
Red 1 = 3 medals
Red 2 = 4 medals
Mode = Color

Shows lots of ghost medals, which patch also fixes.

**Comment by SimpleTease** [comment]
There's also another problem that the original game triggers Victory screen at first player who gets 5 medals. So with Sex mode.

- Human 1 = 3 medals
- Human 2-7 = 0 medals
- Female 8 = 5 medals

Awards screen will show
- Male = 3 medals (White 1)
- Female = 0 medals (White 2)

And then trigger victory screen showing White Female. So patch changes to 3 (Male), 5 (Female) at medals screen to correctly reflect group wins.

**Comment by frranck** [comment]
Are you certain this is not caused by your other PR?

**Comment by SimpleTease** [comment]
I thought I tested on a build months ago (13-09-2020) and got same problems. But that's because Libretro let's you change rules whenever you want during game sessions, which I abuse a lot during local matches. By forcing only during menu, that would solve most bugs I'm seeing. I'll clean everything up and retest.

**Comment by SimpleTease** [comment]
Now targeted for Skynet only.

**Comment by SimpleTease** [comment]
And now I realize I can do more cleanups since medals bugs are gone.

File: ai/MrboomHelper.cpp [hunk 1]
```diff
@@ -447,13 +447,13 @@ int victories(int player)
...
    {
-   case 0:
+   case 0:      // selfie mode
       return(m.victoires[player]);
 
       break;
 
    case 1:      // color mode
-      return(m.victoires[player / 2]);
+      return(m.victoires[player & ~1]);
```
Comment (frranck) [inline ai/MrboomHelper.cpp:457]: so there was a bug here?

**Comment by frranck** [comment]
thanks

**Comment by frranck** [comment]
nice hack

File: ai/MrboomHelper.cpp (hunk 1 above)
Comment (SimpleTease) [reply ai/MrboomHelper.cpp:457]: White = m.victoires[0], m.team[0]
Red = m.victoires[2], m.team[1]
Blue = m.victoires[4], m.team[2]
Green = m.victoires[6], m.team[3]

**Pull request merged**